### DISCUSSION TITLE
What is a sequence expression?

### DISCUSSION THREAD

**Discussion by reagle** [saulpw/visidata#1868] [Q&A]
gz= expr    set current column for selected rows to the items in result of Python sequence expr

**Comment by anjakefala** [comment]
An example is `range(1,100)`! A Python expression that results in a sequence (list, string, etc).

**Reply by reagle** [reply]
@anjakefala, thank you. I'm not sure about the difference between `g=`  and `gz=` though. Is it:

* `g=` can use column names as variables and can return types other than sequence (list, tuple, str, etc.)
* `gz=`cannot use column names and can only return a sequence

?

**Reply by saulpw** [reply]
@reagle So, `=` adds an expression column; you can think of `g=` as doing the same thing "in-place" in the current column (the same is true with `g*`), and only applying to selected rows because that's what `g` does normally.  So the semantics of the expression is the same as with `=`, as you noted.

`gz=` is more closely-related to `z=`, which sets only a single cell to the given Python expression.  If we think about `g` applying to `z=`, then it would be like setting all selected rows to the same Python expression.  But that's what `g=` does already, and so there wouldn't be any difference, unless maybe it computes the value once for the current row, and then sets all selected rows to that same value.  There wasn't any need for that functionality, but a few people asked how they could create a column of integers 1 to N, and so I added `gz=` to take a range object or any iterator/sequence, to fill out the values.

To be honest, the range use case has been largely supplanted by `gi`--and `i` makes it even easier.  I don't think it's used much otherwise.

**Comment by reagle** [comment]
If it isn't used much, I'll remove it from the first 3 decks (where I'm focused). Your point about `g=` does touch on something I had a hard time getting my head around. My "hint" for that card says "relative to z= (current cell), selected cells are more global."
